note.h

```diff
@@ -26,50 +26,47 @@
 #define NOTE_C_VERSION_PATCH 1
 
 // If double and float are the same size, then we must be on a small MCU. Turn
-// on NOTE_LOWMEM to conserve memory.
+// on NOTE_C_LOW_MEM to conserve memory.
 #if defined(FLT_MAX_EXP) && defined(DBL_MAX_EXP)
 #if (FLT_MAX_EXP == DBL_MAX_EXP)
-#define NOTE_LOWMEM
+#define NOTE_C_LOW_MEM
 #endif
 #elif defined(__FLT_MAX_EXP__) && defined(__DBL_MAX_EXP__)
 #if (__FLT_MAX_EXP__ == __DBL_MAX_EXP__)
-#define NOTE_LOWMEM
+#define NOTE_C_LOW_MEM
 #endif
 #else
 #error What are floating point exponent length symbols for this compiler?
 #endif
 
-#ifndef JNUMBER
-#ifdef NOTE_LOWMEM
-#define JNUMBER float
-#else
-#define JNUMBER double
-#endif
-#endif
-
-#ifndef JINTEGER
-#ifdef NOTE_LOWMEM
-#define JINTEGER int32_t
-#define JINTEGER_MIN INT32_MIN
-#define JINTEGER_MAX INT32_MAX
-#else
-#define JINTEGER int64_t
-#define JINTEGER_MIN INT64_MIN
-#define JINTEGER_MAX INT64_MAX
-#endif
+// NOTE_LOWMEM is the old name of NOTE_C_LOW_MEM. If NOTE_C_LOW_MEM is defined,
+// we define NOTE_LOWMEM as well, for backwards compatibility. NOTE_FLOAT is
+// also no longer used internally, but used to be defined when NOTE_LOWMEM was
+// defined. It's also preserved here for backwards compatibility.
+#ifdef NOTE_C_LOW_MEM
+#define NOTE_LOWMEM
```

Comment:
This is backwards.

There is no longer any code gated by `NOTE_LOWMEM`. If `NOTE_LOWMEM` is defined by the build system, then we should define `NOTE_C_LOW_MEM` to reactive all the gates.
```suggestion
#ifdef NOTE_LOWMEM
#define NOTE_C_LOW_MEM
```

Reply:
Once this is addressed :point_up_2:, then I think we are good to merge.